src/Tasks/ResolveKeySource.cs

```diff
@@ -127,7 +127,15 @@ private bool ResolveAssemblyKey()
                             fs = File.OpenRead(KeyFile);
                             int fileLength = (int)fs.Length;
                             var keyBytes = new byte[fileLength];
+#if NET7_0_OR_GREATER
+                            // fail fast in case the file is not read till the end
+                            fs.ReadExactly(keyBytes, 0, fileLength);
+#else
+#pragma warning disable CA2022 // Avoid inexact read with 'Stream.Read'
+                            // TODO: Read the count of read bytes and check if it matches the expected length, if not raise an exception
                             fs.Read(keyBytes, 0, fileLength);
```

Comment:
I'd still prefer a polyfill here so the source can just unconditionally say `ReadExactly`

Reply:
Let's track that than. It feels that it'd be outside of scope of retargetting change.